config/core/test-configs.yaml

```diff
@@ -285,6 +285,12 @@ test_plans:
       kselftest_collections: "arm64"
     filters: *kselftest_no_fragment
 
+  kselftest-bpf:
+    <<: *kselftest
+    params:
+      job_timeout: '10'
+      kselftest_collections: "bpf"
```

Comment:
I don't think we can actually do this because, like you mentioned, BPF requires an unreasonably cutting-edge toolchain.  It's actually one of the reasons why it got disabled by default:
```
commit 7a6eb7c34a78498742b5f82543b7a68c1c443329 (tag: linux-kselftest-fixes-5.11-rc4)
Author: Mark Brown <broonie@kernel.org>
Date:   Thu Dec 10 18:52:33 2020 +0000

    selftests: Skip BPF seftests by default
```

So a prerequisite before being able to run this test automatically would be to have a dedicated job to build a custom toolchain Docker image.  This sounds like something to do with the new API, at some point in 2024 I guess.